src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,232 @@
+package duke.command;
+
+import duke.application.Application;
+import duke.exception.DukeException;
+import duke.exception.InvalidTaskIndexException;
+import duke.exception.NoCommandFoundException;
+import duke.exception.TaskIndexOutOfRangeException;
+import duke.message.Message;
+import duke.task.Task;
+
+/**
+ * Represents an abstract command that interacts with the application with some content string.
+ */
+public abstract class Command {
+    protected String content;
+    protected Command(String content) {
+        this.content = content;
+    }
+
+    /**
+     * Creates a Command instance based on the given command name and content.
+     *
+     * @param commandName The name of the command.
+     * @param content The content associated with the command.
+     * @return A Command instance corresponding to the specified command name.
+     * @throws NoCommandFoundException If the provided command name is not recognized.
+     */
+    public static Command of(String commandName, String content)
+            throws NoCommandFoundException {
+        switch (commandName) {
+        case "bye":
+            return new Bye(content);
+        case "todo":
+            return new Todo(content);
+        case "deadline":
+            return new Deadline(content);
+        case "event":
+            return new Event(content);
+        case "find":
+            return new Find(content);
+        case "list":
+            return new List(content);
+        case "mark":
+            return new Mark(content);
+        case "unmark":
+            return new Unmark(content);
+        case "delete":
+            return new Delete(content);
+        default:
+            throw new NoCommandFoundException(commandName);
+        }
```

Comment:
I like that you followed the coding standard for case and switch which is a common mistake!